buildkite/package_uploads.go

```diff
@@ -0,0 +1,216 @@
+package buildkite
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/buildkite/go-buildkite/v3/internal/bkmultipart"
+)
+
+// CreatePackageInput specifies the input parameters for the Create method.
+// All fields are required, but if PackageFile is an [os.File], Filename can be omitted.
+type CreatePackageInput struct {
+	Package  io.Reader // The package to upload. This can be an [os.File], or any other [io.Reader].
+	Filename string    // The name of the file to upload. If PackageFile is an [os.File], this can be omitted, and the file's name will be used.
+}
+
+// Create creates a package in a registry for an organization
+func (ps *PackagesService) Create(ctx context.Context, organizationSlug, registrySlug string, cpi CreatePackageInput) (Package, *Response, error) {
+	filename := cpi.Filename
+	if f, ok := cpi.Package.(*os.File); ok && filename == "" {
+		filename = f.Name()
+	}
+
+	packageTempFile, err := normalizeToFile(cpi.Package, filename)
+	if err != nil {
+		return Package{}, nil, fmt.Errorf("writing package to tempfile: %v", err)
+	}
+	defer func() {
+		packageTempFile.Close()
+		os.Remove(packageTempFile.Name())
+	}()
+
+	s := bkmultipart.NewStreamer()
+	s.WriteFile(fileFormKey, packageTempFile, filename)
+
+	ppu, _, err := ps.RequestPresignedUpload(ctx, organizationSlug, registrySlug)
+	if err != nil {
+		return Package{}, nil, fmt.Errorf("requesting presigned upload: %v", err)
+	}
+
+	s3URL, err := ppu.Perform(ctx, ps, packageTempFile)
+	if err != nil {
+		return Package{}, nil, fmt.Errorf("performing presigned upload: %v", err)
+	}
+
+	p, resp, err := ppu.Finalize(ctx, ps, s3URL)
+	if err != nil {
+		return Package{}, nil, fmt.Errorf("finalizing package: %v", err)
+	}
+
+	return p, resp, nil
```

Comment:
What is the response return used for? Looks like it is ignored when it is called below. Perhaps delete it?

If `Create` is implementing a "workflow" (i.e. sequence of HTTP requests and responses) then is expecting the caller to do something with the `Response` as well as the `Packge` leaking the abstraction a bit?

Reply:
i _think_ for most use cases, the response returned can be used for more inspection into errors/deeper introspection of HTTP.... stuff. to be honest, i only added it to this method because every other public method in this module returns a tuple of `[T any](T, *Response, error)`, and it felt like it'd be weird to have one method that doesn't do that?'

`Create` is indeed implementing a workflow - you might be right about making it just return `(Package, error)`